pyaci/core.py

```diff
@@ -195,14 +204,21 @@ def __init__(self, url, session=None, verify=False, disableWarnings=True,
         self._wsMos = defaultdict(deque)
         self._wsReady = Event()
         self._wsEvents = {}
+        self._autoRefresh = False
+        self._autoRefreshThread = None
+        self._login = {}
+        self._autotest = False
 
     @property
     def session(self):
         return self._session
 
     @property
     def webSocketUrl(self):
-        token = self._rootApi()._session.cookies['APIC-cookie']
+        if 'APIC-cookie' in self._rootApi()._session.cookies :
+            token = self._rootApi()._session.cookies['APIC-cookie']
+        else:
+            raise Exception('APIC-cookie NOT found.. Make sure you have logged in.')
```

Comment:
We should have specific exceptions/errors thrown instead of the blanket Exception. We can handle this later.